op-node/rollup/derive/engine_queue.go

```diff
@@ -564,8 +564,14 @@ func (eq *EngineQueue) forceNextSafeAttributes(ctx context.Context) error {
 			// block is somehow invalid, there is nothing we can do to recover & we should exit.
 			// TODO: Can this be triggered by an empty batch with invalid data (like parent hash or gas limit?)
 			if len(attrs.Transactions) == depositCount {
-				eq.log.Error("deposit only block was invalid", "parent", eq.safeAttributes.parent, "err", err)
-				return NewCriticalError(fmt.Errorf("failed to process block with only deposit transactions: %w", err))
+				switch depositCount {
+				case 0:
+					eq.log.Error("empty payload was invalid", "parent", eq.safeAttributes.parent, "err", err)
+					return NewCriticalError(fmt.Errorf("failed to process block with 0 transactions: %w", err))
```

Comment:
Why would we ever get a block with no transactions? Every block must have at least the L1 info deposit transaction. Given at this point we know that at this point we have the expected number of deposit transactions, the payload must not be empty.

Did we actually hit this case at some point?

Reply:
We can close this, added as an ask in https://github.com/ethereum-optimism/optimism/issues/8993. There was an issue in `op-reth` with some recent reorganization, where the `optimism` feature flag wasn't being enabled for one of the dependencies. Solved in https://github.com/paradigmxyz/reth/pull/6065.

Reply:
Actually I was wrong - I'd misread the situation and thought we were computing the expected deposit count from what we sent to the EL, but we are just counting the deposits in the payload we have.  So it could indeed be 0.  I'd probably still lean towards keeping this code simple but could rephrase to something like "failed to process block with no non-deposit transactions".